### PR TITLE
Shale reference compositions and McLennan 2001 crustal compositions

### PR DESCRIPTION
**Pull request by kaarelmand**
Some new, mostly shale-based reference compositions for pyrolite.

## Description
This PR adds some composition .csv files to the [refcomp](https://github.com/morganjwilliams/pyrolite/tree/master/pyrolite/data/geochem/refcomp) folder, mainly those concerning shales and other sediments.

- [x] Phanerozoic cratonic shale compilation by Condie (1993) -- PHS_Condie1993.csv
- [x] Post-Archean Australian Shale (PAAS) by Taylor & McLennan (1985), updates by McLennan (2001) -- PAAS_TaylorMcLennan1985.csv
- [x] North American Composite Shale (NASC) by Gromet et al. (1984), updates by Taylor & McLennan (1985), Condie, 1993, McLennan, 2001 -- NASC_Gromet1984.csv
- [x] Upper continental crust, lower continental crust and bulk continental crust composition estimates by McLennan (2001) -- UCC_McLennan2001.csv, LCC_McLennan2001.csv, BCC_McLennan2001.csv
- [x] Proterozoic and Archean cratonic shales by Condie (1993) -- PRS_Condie1993.csv, ARS_Condie1993.csv
- [x] "PAAS," being a smoother REY dataset from a different set of Australian shales by Pourmand et al. (2012) -- PAAS_Pourmand2012.csv
- [x] European Shale (EUS) by Bau et al. (2018) -- EUS_Bau2018.csv
- [x] MUd from Queensland (MUQ) by Kamber et al. (2005), being a semi-popular proxy for recent average detritus -- MUQ_Kamber2005.csv 

## Related Issue
Addresses point 2 of issue #35.

## Motivation and Context
Pyrolite was previously missing widely-cited shale reference compositions, most notably PAAS. This PR adds those to pyrolite's already extensive reference composition database.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by kaarelmand** [comment]
While I did add all these files to the refcomp folder, I am not sure if I have to run any scripts to re-populate the `refdb.json` file.

Nor am I sure how to update the documentation -- it seems to me that the page on the reference compositions is automatically populated?

**Comment by morganjwilliams** [comment]
Thanks @kaarelmand, looks great! 

I'll have a quick look at the CSVs, but I suspect this should be able to be merged without any issues. If you wanted to check they all import nicely, you could run the `pyrolite.geochem` tests locally (there's one included for the database update) - although note that the Travis build for the PR passed, and I can confirm that tests pass on my machine too!

To round off the PR, could you add a commit after running the `update_database()` function locally to add you new files to the `refdb.json` file? This is where `pyrolite` will look for the reference compositions.

**Comment by morganjwilliams** [comment]
On a slight tangent, I have a feeling that soon it will be worth writing up a tutorial about the variety of reference compositions, and their general applicability. Let me know if you're happy to contribute some info for a shale section!

**Comment by morganjwilliams** [comment]
The documentation example does indeed have an auto-documented list of the reference compositions, so there's no need to update this.

Given we now have 31 compositions, this will be something to break out into a new page soon. I've added #38 to this end, and we can address that separately.

**Comment by kaarelmand** [comment]
I'm in the process of fixing the earlier .csv's I'd uploaded. I also added a todo list in the description.

**Comment by kaarelmand** [comment]
Some musings, so I don't forget:

I'm thinking I should probably hold off on adding the conglomerates and sandstones from Condie, 1993 -- I don't know/think that anyone would actually need those compositions. And if someone does, it's very easy to add them in. Seems a bit silly in hindsight.

For the European shale -- Bau et al. only report REY values, but there's some older data floating around for other elements on the same sample set.

Sometime in the future, it might be worth thinking of a function that would take a time-series of element concentration data over Earth history, and normalize it to the changing crust in the Archean--Proterozoic. Condie, 1993, actually has 7 different composition brackets through Earth history, even though I only included his eon-averages.

**Comment by kaarelmand** [comment]
All right, that should be it for this PR! Not sure if I still have to run any tests or whatnot?

**Comment by morganjwilliams** [comment]
There's no specific tests for the data files, but I'll pull down the fork and have a look locally.

File: pyrolite/data/geochem/refcomp/PAAS_TaylorMcLennan1985.csv [hunk 1]
```diff
@@ -0,0 +1,95 @@
+var,value,units,unc_2sigma,constraint_var,constraint_value,constraint_comment
+ModelName,PAAS_TM1985,,,,,
+Reservoir,PostArcheanAustralianShale,,,,,
+ModelType,Average,,,,,
+Reference,Taylor & McLennan 1985, McLennan 2001,,,,
```
Comment (morganjwilliams) [inline pyrolite/data/geochem/refcomp/PAAS_TaylorMcLennan1985.csv:5]: The reference here is split across the value and and units columns - add them to the first with a semicolon separator.

File: pyrolite/data/geochem/refcomp/NASC_Gromet1984.csv [hunk 2]
```diff
@@ -0,0 +1,95 @@
+var,value,units,unc_2sigma,constraint_var,constraint_value,constraint_comment
+ModelName,NASC_Gromet1984,,,,,
+Reservoir,NorthAmericanShaleComposite,,,,,
+ModelType,Average,,,,,
+Reference,"Gromet et al., 1984; Taylor and McLennan, 1985; Condie, 1993; McLennan, 2001",,,,,
```
Comment (morganjwilliams) [inline pyrolite/data/geochem/refcomp/NASC_Gromet1984.csv:5]: In the references for this file there's string which contains the file delimiter (a comma) - perhaps change this to remove commas and separate references with semicolons.

File: files.txt [hunk 3]
```diff
@@ -0,0 +1,231 @@
+/home/kaarel/anaconda3/lib/python3.7/site-packages/pyrolite-0.2.5+31.g13bb5cc-py3.7.egg/EGG-INFO/SOURCES.txt
```
Comment (morganjwilliams) [inline files.txt:1]: I think this was inadvertently committed to the branch and should be removed.

**Comment by morganjwilliams** [comment]
@kaarelmand if you could fix the two reference cells for the NASC and PAAS data files I mentioned above and remove the extra text file I think this will then be ready to merge. I'll update the refdb.json after this - have an issue on my side which is likely due to a new version of TinyDB, and I'll update develop to fix this a the same time.

**Comment by kaarelmand** [comment]
Whoops, these issues should be fixed now!

**Comment by kaarelmand** [comment]
Whoops, these issues should be fixed now!

**Comment by morganjwilliams** [comment]
Thanks for this one @kaarelmand! I had to re-update the reference database to pass the tests but everything else looks good. You can see some of the new compositions pop up in the [normalization example](https://pyrolite.readthedocs.io/en/develop/examples/geochem/normalization.html).

**Pull request merged**